### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Bludidi**
***Hello Reviewer!***

## In this milestone we have implemented the following: 
- [ ] Updated the `ApplicationController`
- [ ] Installed `devise gem`
- [ ] Verified if we can: 
   - Register a new user.
   - User logs in with a combination of `email` and `password`
   - Hashed password is stored in the database. 
   - Ask for confirmation of an email.
   - User can reset password.

### CODE REVIEW

File: db/migrate/20230420114025_add_unique_index_to_users_email.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToUsersEmail < ActiveRecord::Migration[7.0]
+  def change
+    add_index :users, :email, unique: true, where: "email IS NOT NULL AND email != ''"
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline db/migrate/20230420114025_add_unique_index_to_users_email.rb]: - [ ] This is actually causing an error during migration kindly remove as email is already unique in the previous migration file

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+    <div class="field input-div">
+      <%= f.label :email %><br />
+      <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+    </div>
+
+    <div class="field input-div">
+      <%= f.label :password %><br />
+      <%= f.password_field :password, autocomplete: "current-password" %>
```
Comment (VICTOR-CHUKWU) [inline app/views/devise/sessions/new.html.erb]: - [ ] after regristation i could not login in kindly check this [video](https://www.loom.com/share/1bc7e689f2934eb78c41f6123beed6d9) out

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Bludidi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @Bludidi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

well done


## Highlights
:heavy_check_mark: Nice UI
:heavy_check_mark: Great effort in making the requested changes 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**